src/Sentry/SentrySdk.cs

```diff
@@ -240,6 +240,17 @@ public void Dispose()
     [DebuggerStepThrough]
     public static IDisposable PushScope<TState>(TState state) => CurrentHub.PushScope(state);
 
+    /// <summary>
+    /// Restores a previously saved/cloned scope to the top of the stack
+    /// </summary>
+    public static void RestoreScope(Scope savedScope)
```

Comment:
It feels backwards that we're adding a public API to the static class but not exposing that on `ISentryClient` or `IHub`. This static API was designed for convinience, and only mapping to one of the two interfaces. And code we write relies on the abstractions for testability sake.

We can see this design choice is odd when we're doing type check for a specific type of Hub before we can call the method.

Can we avoid these public methods altogether?